src/main/java/subway/path/Path.java

```diff
@@ -0,0 +1,83 @@
+package subway.path;
+
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+import subway.section.Section;
+import subway.station.Station;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Path {
+    private final List<Station> stations;
+    private final long distance;
+
+    public Path(List<Section> sections, Station start, Station end) {
+        WeightedMultigraph<Station, DefaultWeightedEdge> graph
+                = new WeightedMultigraph<>(DefaultWeightedEdge.class);
+        init(graph, sections);
+        this.stations = getStations(graph, start, end);
+        this.distance = calcDistance(graph, start, end);
+    }
```

Comment:
아래와 같이 getStations와 calcDistance 메서드 분리 없이도 구성할 수 도 있겠네요 :) 

```suggestion
    public Path(List<Section> sections, Station start, Station end) {
        WeightedMultigraph<Station, DefaultWeightedEdge> graph
                = new WeightedMultigraph<>(DefaultWeightedEdge.class);
        init(graph, sections);
        
        validateIsConnected(graph, start, end);
        validateIsDifferentStation(start, end);
        
        GraphPath<Station, DefaultWeightedEdge> path = new DijkstraShortestPath<>(graph).getPath(start, end);
        
        this.stations = path.getVertexList();
        this.distance = (long) path.getWeight();
    }
```